clang/lib/DPCT/SaveNewFiles.cpp

```diff
@@ -505,6 +511,25 @@ int writeReplacementsToFiles(
       continue;
     }
 
+    // Check for another file with SYCL extension. For example
+    // In in-root we have
+    //  * src.cpp
+    //  * src.cu
+    // After migration we will end up replacing src.cpp with migrated src.cu
+    // when the --sycl-file-extension is cpp
+    // In such a case warn the user.
+    auto FE = Rewrite.getSourceMgr().getFileManager().getFile(
+        MigFilepathNoOutRoot, false, false);
+
+    if (FE && !(FileType & SPT_CppHeader)) {
+      std::string Warning =
+          "[WARNING] Replacing an existing file '" + MigFilepathNoOutRoot +
+          "' with a migrated file because they share the same "
+          "extension. To use a different extension see option "
+          "--sycl-file-extension\n";
+      PrintMsg(Warning);
```

Comment:
If the following behavior is unsafe, should we directly return here?

Reply:
Not returning from here have intendent behaviour! As in, the migrated file is never overwritten by an existing file. Please see last two lines in the output below.

```
The directory "dpct_output" is used as "out-root"
Parsing: /export/users/dhr/cuda_samples/mig_file_extn/main.cu
Parsing: /export/users/dhr/cuda_samples/mig_file_extn/src.cu
NOTE: /export/users/dhr/cuda_samples/mig_file_extn/src.cpp is treated as a CUDA file by default. Use the --extra-arg=-xc++ option to treat /export/users/dhr/cuda_samples/mig_file_extn/src.cpp as a C++ file if needed.
Parsing: /export/users/dhr/cuda_samples/mig_file_extn/src.cpp
Analyzing: /export/users/dhr/cuda_samples/mig_file_extn/main.cu
Analyzing: /export/users/dhr/cuda_samples/mig_file_extn/src.cu
Analyzing: /export/users/dhr/cuda_samples/mig_file_extn/src.cpp
Migrating: /export/users/dhr/cuda_samples/mig_file_extn/main.cu
Migrating: /export/users/dhr/cuda_samples/mig_file_extn/src.cu
Migrating: /export/users/dhr/cuda_samples/mig_file_extn/src.cpp
Saved new version of /export/users/dhr/cuda_samples/mig_file_extn/dpct_output/header.hpp.yaml file
[WARNING] Replacing an existing file '/export/users/dhr/cuda_samples/mig_file_extn/header.hpp' with a migrated file because they share the same extension. To use a different extension see option --sycl-file-extension
[WARNING] Replacing an existing file '/export/users/dhr/cuda_samples/mig_file_extn/src.cpp' with a migrated file because they share the same extension. To use a different extension see option --sycl-file-extension
Processed 3 file(s) in -in-root folder "/export/users/dhr/cuda_samples/mig_file_extn"

See Diagnostics Reference to resolve warnings and complete the migration:
https://www.intel.com/content/www/us/en/docs/dpcpp-compatibility-tool/developer-guide-reference/current/diagnostics-reference.html
[Warning] Error happened while formatting. Generating unformatted code.
File '/export/users/dhr/cuda_samples/mig_file_extn/dpct_output/header.hpp' already exists; skipping it.
File '/export/users/dhr/cuda_samples/mig_file_extn/dpct_output/src.cpp' already exists; skipping it.
```

To try, I added `continued` (as this is a part of loop)! But this results in format failure. See the last warning in the output below.
```
The directory "dpct_output" is used as "out-root"
Parsing: /export/users/dhr/cuda_samples/mig_file_extn/main.cu
Parsing: /export/users/dhr/cuda_samples/mig_file_extn/src.cu
NOTE: /export/users/dhr/cuda_samples/mig_file_extn/src.cpp is treated as a CUDA file by default. Use the --extra-arg=-xc++ option to treat /export/users/dhr/cuda_samples/mig_file_extn/src.cpp as a C++ file if needed.
Parsing: /export/users/dhr/cuda_samples/mig_file_extn/src.cpp
Analyzing: /export/users/dhr/cuda_samples/mig_file_extn/main.cu
Analyzing: /export/users/dhr/cuda_samples/mig_file_extn/src.cu
Analyzing: /export/users/dhr/cuda_samples/mig_file_extn/src.cpp
Migrating: /export/users/dhr/cuda_samples/mig_file_extn/main.cu
Migrating: /export/users/dhr/cuda_samples/mig_file_extn/src.cu
Migrating: /export/users/dhr/cuda_samples/mig_file_extn/src.cpp
Saved new version of /export/users/dhr/cuda_samples/mig_file_extn/dpct_output/header.hpp.yaml file
[WARNING] Replacing an existing file '/export/users/dhr/cuda_samples/mig_file_extn/header.hpp' with a migrated file because they share the same extension. To use a different extension see option --sycl-file-extension
[WARNING] Replacing an existing file '/export/users/dhr/cuda_samples/mig_file_extn/src.cpp' with a migrated file because they share the same extension. To use a different extension see option --sycl-file-extension
Processed 3 file(s) in -in-root folder "/export/users/dhr/cuda_samples/mig_file_extn"

See Diagnostics Reference to resolve warnings and complete the migration:
https://www.intel.com/content/www/us/en/docs/dpcpp-compatibility-tool/developer-guide-reference/current/diagnostics-reference.html
[Warning] Error happened while formatting. Generating unformatted code.
File '/export/users/dhr/cuda_samples/mig_file_extn/dpct_output/header.hpp' already exists; skipping it.
File '/export/users/dhr/cuda_samples/mig_file_extn/dpct_output/src.cpp' already exists; skipping it.
```

Please suggest what might be the correct thing to do.